Rtabmap is not compatible with Ubuntu 18.04 PCL 1.11 in ROS melodic

My environment is Ubuntu 18.04, PCL 1.11. The code works with Ubuntu 16.04, PCL 1.8. Can you please confirm this bug?

The error pops up like following: 

/home/chaonew/catkin_ws/src/rtabmap_ros/src/OdometryROS.cpp: In member function ‘void rtabmap_ros::OdometryROS::processData(const rtabmap::SensorData&, const ros::Time&)’:
/home/chaonew/catkin_ws/src/rtabmap_ros/src/OdometryROS.cpp:735:45: error: call of overloaded ‘PointXYZRGB(int, int, int)’ is ambiguous
     pcl::PointXYZRGB pt(inlier?0:255, 255, 0);
                                             ^
In file included from /usr/local/include/pcl-1.11/pcl/point_types.h:349:0,
                 from /usr/local/include/pcl-1.11/pcl/common/impl/copy_point.hpp:40,
                 from /usr/local/include/pcl-1.11/pcl/common/copy_point.h:58,
                 from /usr/local/include/pcl-1.11/pcl/common/impl/io.hpp:44,
                 from /usr/local/include/pcl-1.11/pcl/common/io.h:587,
                 from /usr/local/include/pcl-1.11/pcl/io/file_io.h:41,
                 from /usr/local/include/pcl-1.11/pcl/io/pcd_io.h:45,
                 from /home/chaonew/catkin_ws/src/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:70,
                 from /home/chaonew/catkin_ws/src/rtabmap_ros/src/OdometryROS.cpp:35:
/usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:641:12: note: candidate: pcl::PointXYZRGB::PointXYZRGB(float, float, float)
     inline PointXYZRGB (float _x, float _y, float _z):
            ^~~~~~~~~~~
/usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:638:12: note: candidate: pcl::PointXYZRGB::PointXYZRGB(uint8_t, uint8_t, uint8_t)
     inline PointXYZRGB (std::uint8_t _r, std::uint8_t _g, std::uint8_t _b):
            ^~~~~~~~~~~
rtabmap_ros/CMakeFiles/rtabmap_ros.dir/build.make:110: recipe for target 'rtabmap_ros/CMakeFiles/rtabmap_ros.dir/src/OdometryROS.cpp.o' failed
make[2]: *** [rtabmap_ros/CMakeFiles/rtabmap_ros.dir/src/OdometryROS.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:8378: recipe for target 'rtabmap_ros/CMakeFiles/rtabmap_ros.dir/all' failed
make[1]: *** [rtabmap_ros/CMakeFiles/rtabmap_ros.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2



                                             ^

I pulled the code to the newest version, there is still some bug like following:

error: no matching function for call to ‘rtabmap::Rtabmap::getSignatureCopy(int&, rtabmap_ros::GetNodeDataRequest_<std::allocator<void> >::_images_type&, rtabmap_ros::GetNodeDataRequest_<std::allocator<void> >::_scan_type&, rtabmap_ros::GetNodeDataRequest_<std::allocator<void> >::_user_data_type&, rtabmap_ros::GetNodeDataRequest_<std::allocator<void> >::_grid_type&)’
   Signature s = rtabmap_.getSignatureCopy(id, req.images, req.scan, req.user_data, req.grid);

In file included from /home/chaonew/catkin_ws/src/rtabmap_ros/include/rtabmap_ros/CoreWrapper.h:49:0,
                 from /home/chaonew/catkin_ws/src/rtabmap_ros/src/CoreWrapper.cpp:28:
/usr/local/lib/rtabmap-0.20/../../include/rtabmap-0.20/rtabmap/core/Rtabmap.h:183:12: note: candidate: rtabmap::Signature rtabmap::Rtabmap::getSignatureCopy(int, bool, bool, bool, bool, bool, bool) const
  Signature getSignatureCopy(int id, bool images, bool scan, bool userData, bool occupancyGrid, bool withWords, bool withGlobalDescriptors) const;
            ^~~~~~~~~~~~~~~~
/usr/local/lib/rtabmap-0.20/../../include/rtabmap-0.20/rtabmap/core/Rtabmap.h:183:12: note:   candidate expects 7 arguments, 5 provided
rtabmap_ros/CMakeFiles/rtabmap_sync.dir/build.make:230: recipe for target 'rtabmap_ros/CMakeFiles/rtabmap_sync.dir/src/CoreWrapper.cpp.o' failed
make[2]: *** [rtabmap_ros/CMakeFiles/rtabmap_sync.dir/src/CoreWrapper.cpp.o] Error 1
CMakeFiles/Makefile2:6867: recipe for target 'rtabmap_ros/CMakeFiles/rtabmap_sync.dir/all' failed
make[1]: *** [rtabmap_ros/CMakeFiles/rtabmap_sync.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

--- a/src/CoreWrapper.cpp
+++ b/src/CoreWrapper.cpp
@@ -1883,10 +1883,10 @@ void CoreWrapper::process(
 				if(maxMappingNodes_ > 0 && filteredPoses.size()>1)
 				{
 					std::map<int, Transform> nearestPoses;
-					std::vector<int> nodes = graph::findNearestNodes(filteredPoses, mapToOdom_*odom, maxMappingNodes_);
-					for(std::vector<int>::iterator iter=nodes.begin(); iter!=nodes.end(); ++iter)
+					std::map<int, float> nodes = graph::findNearestNodes(filteredPoses, mapToOdom_*odom, maxMappingNodes_);
+					for(std::map<int, float>::iterator iter=nodes.begin(); iter!=nodes.end(); ++iter)
 					{
-						std::map<int, Transform>::iterator pter = filteredPoses.find(*iter);
+						std::map<int, Transform>::iterator pter = filteredPoses.find(iter->first);
 						if(pter != filteredPoses.end())
 						{
 							nearestPoses.insert(*pter);
@@ -2770,10 +2770,10 @@ void CoreWrapper::publishMapCallback(
 			if(maxMappingNodes_ > 0 && filteredPoses.size()>1)
 			{
 				std::map<int, Transform> nearestPoses;
-				std::vector<int> nodes = graph::findNearestNodes(filteredPoses, filteredPoses.rbegin()->second, maxMappingNodes_);
-				for(std::vector<int>::iterator iter=nodes.begin(); iter!=nodes.end(); ++iter)
+				std::map<int, float> nodes = graph::findNearestNodes(filteredPoses, filteredPoses.rbegin()->second, maxMappingNodes_);
+				for(std::map<int, float>::iterator iter=nodes.begin(); iter!=nodes.end(); ++iter)
 				{
-					std::map<int, Transform>::iterator pter = filteredPoses.find(*iter);
+					std::map<int, Transform>::iterator pter = filteredPoses.find(iter->first);
 					if(pter != filteredPoses.end())
 					{
 						nearestPoses.insert(*pter);

--- a/src/OdometryROS.cpp
+++ b/src/OdometryROS.cpp
@@ -654,7 +654,9 @@ void OdometryROS::processData(const SensorData & data, const rclcpp::Time & stam
 			for(std::map<int, cv::Point3f>::const_iterator iter=info.localMap.begin(); iter!=info.localMap.end(); ++iter)
 			{
 				bool inlier = info.words.find(iter->first) != info.words.end();
-				pcl::PointXYZRGB pt(inlier?0:255, 255, 0);
+				pcl::PointXYZRGB pt;
+				pt.r = inlier?0:255;
+				pt.g = 255;
 				pt.x = iter->second.x;
 				pt.y = iter->second.y;
 				pt.z = iter->second.z;

